core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -228,15 +241,23 @@ func (eb *EthBroadcaster) newResendBackoff() backoff.Backoff {
 }
 
 func (eb *EthBroadcaster) monitorEthTxs(k ethkey.State, triggerCh chan struct{}) {
+	defer eb.wg.Done()
+
 	ctx, cancel := utils.ContextFromChan(eb.chStop)
 	defer cancel()
 
+	if eb.autoSyncNonce {
+		eb.logger.Debugw("Auto-syncing nonce", "address", k.Address)
+		eb.SyncNonce(ctx, k)
```

Comment:
Maybe worth checking ctx explicitly to short circuit rather than running `ProcessUnstartedEthTxs`?
```suggestion
		eb.SyncNonce(ctx, k)
		if ctx.Err() != nil {
			return
		}
```